Week 1/Weekly Practice Problems (Week 1)/1365: How Many Numbers Are Smaller Than the Current Number/dominic.cpp

```diff
@@ -0,0 +1,13 @@
+class Solution {
+public:
+    vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
+        int n = *max_element(nums.begin(), nums.end()) + 2;
+        vector<int> counts(n, 0);
+
+        for (int x : nums) counts[x+1]++;
+        for (int i = 1; i < n; i++) counts[i] += counts[i-1];
+        for (auto &x : nums) x = counts[x];
```

Comment:
hmmm, this solution got accepted?

wont this include things that are after the current num?

Reply:
Not sure if I understand what you mean 🤔 
Line 7 is storing the count of each x in 'x+1', so we don't count duplicates.
Line 8 does a prefix sum, which will only increase if we've increased the count at this index when line 7 ran.
Line 9 takes the stored count of x and replaces x with it.

Which line might be including things after the current num?

Reply:
when we do L9, the prefix sum in L8 has information from the entire `nums` array though?

e.g. `[3,4,2]` at 4, wont this impl count for both 3, and, 2?

Reply:
Ohh, I think I get what you're saying. Here's the problem description:
Given the array nums, for each nums[i] find out how many numbers in the array are smaller than it. That is, for each nums[i] you have to count the number of valid j's such that j != i and nums[j] < nums[i].

Since it says nums[j] < nums[I] and not j < i, I believe this means the numbers can be counted after the current index

Reply:
oh lol nvm. I mixed this up with another problem